File: src/API/Microsservices/Account/Sonorus.Account.Application/Commands/UpdateUser/UpdateUserInputModel.cs
namespace Sonorus.Account.Application.Commands.UpdateUser;

public class UpdateUserInputModel(string fullname, string nickname, string email) {
    public string Fullname { get; private set; } = fullname;
    public string Nickname { get; private set; } = nickname;
    public string Email { get; private set; } = email;
}

File: src/app/lib/src/dtos/input_models/update_user_input_model.dart
import "dart:convert";

class UpdateUserInputModel {
  final String fullname;
  final String nickname;
  final String email;

  UpdateUserInputModel({
    required this.fullname,
    required this.nickname,
    required this.email
  });

  String toJson() => json.encode({
    "fullname": fullname,
    "nickname": nickname,
    "email": email
  });
}